### DISCUSSION TITLE
Configure retention policy when deploying on Docker

### DISCUSSION THREAD

**Discussion by MPapst** [datalust/seq-tickets#1458] [Ideas]
Is there any possibility to configure the retention policy from the "outside"?

Scenario: I am running in docker and want to have the container running for a test system for several months. To avoid configuration effort, I want to supply e.g. the retention policy in the container configuration.
This would allow me to deploy my test system on the fly.

**Comment by nblumhardt** [comment]
Hi Marco, thanks for the note. Not currently - the first step in improving the Docker deployment story will be https://github.com/datalust/seq-tickets/issues/791 - which should set the overall direction for this. I will close this ticket and make a note over there so that we can come back to it. Thanks again!

**Comment by corentinaltepe** [comment]
Hello. Since #791 is now closed, is there any chance to re-open this ticket? I am looking for a way to apply a default retention policy in docker-compose. Thank you.

**Comment by nblumhardt** [comment]
Thanks for the nudge, @corentinaltepe; [init scripts](docs.datalust.co/docs/init-scripts) give us some new options for enabling this.

The `seqcli` executable is already in the container, and `seqcli retention create` does at least support automated creation of an "all events" property, so the key will be to run (some variety) of init script with a connected/configured `seqcli` on the path 🤔

**Comment by nblumhardt** [comment]
Looping back to this, running a child script with `& disown` may be one way to achieve this:

https://unix.stackexchange.com/a/363217

Other than this kind of solution, having your deployment server run the post-init script once the container is up could be more pleasant/ergonomic.

While we figure out what kind of approach will work best for this, I'll move this thread to a discussion. Happy to help work through it, if anyone's aiming to implement this with Seq today.
